src/main/java/com/faforever/client/preferences/ForgedAlliancePrefs.java

```diff
@@ -41,12 +42,18 @@
   @Deprecated
   private final ObjectProperty<Path> path;
   private final ObjectProperty<Path> installationPath;
-  private final ObjectProperty<Path> customMapsDirectory;
   private final ObjectProperty<Path> preferencesFile;
   private final ObjectProperty<Path> officialMapsDirectory;
-  private final ObjectProperty<Path> modsDirectory;
+  private final ObjectProperty<Path> vaultBaseDirectory;
+  private transient final ObjectProperty<Path> customMapsDirectory;
+  private transient final ObjectProperty<Path> modsDirectory;
   private final IntegerProperty port;
   private final BooleanProperty autoDownloadMaps;
+  /**
+   * Saves if the client checked for special cases in which it needs to set the fallback vault location. See {@link
+   * com.faforever.client.vault.VaultFileSystemLocationChecker}
+   */
+  private final BooleanProperty vaultCheckDone;
```

Comment:
What if user moves stuff to a OneDrive? Is vault path check costly?

Reply:
I am not sure what u mean... If One Drive is installed (fully) it moves Documents to One Dive but the game still uses the old Documents folder but Windows tells the client the folder is in One Drive therefore the game does not find the vault... I tried finding a way of querying the none OneDrive Documents location but because I have no system to try if it works I have not changed that so far. For now we just use the fall back location if the user uses One Drive